### PR TITLE
feat: allow admins to turn off site alerts for patch release versions

### PR DESCRIPTION
**Pull request by dadlerj**
fixes https://github.com/sourcegraph/sourcegraph/issues/3615

That basic semver checking is a bit nasty, I know, but this isn't critical functionality, and I don't want to add a new npm dependency we don't need. Thoughts?

### CODE REVIEW

File: web/src/global/GlobalAlerts.tsx [hunk 1]
```diff
@@ -57,7 +57,14 @@ export class GlobalAlerts extends React.PureComponent<Props, State> {
                         {this.props.isSiteAdmin &&
                             this.state.siteFlags.updateCheck &&
                             !this.state.siteFlags.updateCheck.errorMessage &&
-                            this.state.siteFlags.updateCheck.updateVersionAvailable && (
+                            this.state.siteFlags.updateCheck.updateVersionAvailable &&
+                            ((isSettingsValid<Settings>(this.props.settingsCascade) &&
+                                (this.props.settingsCascade.final['alerts.showMinorUpdates'] === undefined ||
+                                    this.props.settingsCascade.final['alerts.showMinorUpdates'])) ||
```
Comment (felixfbecker) [inline web/src/global/GlobalAlerts.tsx]: I'd use `this.props.settingsCascade.final['alerts.showMinorUpdates'] !== false`, less code and also handles `null`.

File: web/src/global/GlobalAlerts.tsx [hunk 2]
```diff
@@ -114,3 +121,14 @@ export class GlobalAlerts extends React.PureComponent<Props, State> {
...
+
+function isMajorUpdateAvailable(currentVersion: string, updateVersion: string): boolean {
+    const cv = currentVersion.split('.')
+    const uv = updateVersion.split('.')
+    // If either current or update versions aren't semvers (e.g., a user is on a date-based build version),
+    // always return true.
+    if (cv.length !== 3 || uv.length !== 3) {
+        return true
+    }
+    return parseInt(uv[0], 10) !== parseInt(cv[0], 10)
```
Comment (felixfbecker) [inline web/src/global/GlobalAlerts.tsx]: I'm a bit worried about this causing a bug in the future with various valid semver versions, e.g. prereleases. The official `semver` parser is 4.5KB minified+gzipped: https://bundlephobia.com/result?p=semver@6.0.0
and we already have it in our dep tree anyway

**Comment by dadlerj** [comment]
Thanks @felixfbecker! Let me know what you think (also about the strange changes to yarn.lock here, which I also asked about at https://sourcegraph.slack.com/archives/CHXHX7XAS/p1557249936001900)

File: web/src/global/GlobalAlerts.tsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { parseISO } from 'date-fns'
 import differenceInDays from 'date-fns/differenceInDays'
 import * as React from 'react'
 import { Subscription } from 'rxjs'
+import { major, minor, parse } from 'semver'
```
Comment (felixfbecker) [inline web/src/global/GlobalAlerts.tsx]: I would recommend using a `*` import here to make it clear what these functions are parsing at the call site
```suggestion
import * as semver from 'semver'
```

**Pull request merged**